### DISCUSSION TITLE
poetry install location

### DISCUSSION THREAD

**Discussion by blnprasad** [python-poetry/poetry#6418] [General] [answered]
I'm trying to build docker with python3.9 and poetry1.2 version.  I'm installing packages using below command.
       poetry config virtualenvs.create false && poetry install --without dev

looks like poetry installing packages under directory ( ./usr/local/lib/python3.8/dist-packages/httpx) and
my sys path is coming as below
sys.path = [
    '/',
    '/usr/lib/python39.zip',
    '/usr/lib/python3.9',
    '/usr/lib/python3.9/lib-dynload',
    '/root/.local/lib/python3.9/site-packages',
    '/usr/local/lib/python3.9/dist-packages',
    '/usr/lib/python3/dist-packages',
]

since installed directory is not under sys path the packages being ignored. i didn't see this issue with 3.8(/usr/local/lib/python3.8/dist-packages/). can you help what's going wrong here?

**Comment by neersighted** [comment]
When you install with Poetry into system site-packages, the site-packages selected by Poetry will correspond to the interpreter running Poetry.

Try running `poetry debug info` and you should discover which interpreter Poetry is running with.

Note that `virtualenvs.create` is usually a footgun and you do not want to mix your project in with the system Python packages -- you may break your package manager, break Poetry (if installed through OS packaging), experience instability as Poetry is not the only thing managing your Python environment, or run into OS-supplied packages that are incompatible with the Python packaging ecosystem.

What problem are you trying to install by installing into system site-packages?

**Reply by blnprasad** [reply]
I went with that option since it docker container to run my server alone. i'll check more on that

**Comment by neersighted** [comment, marked as answer]
Installation into system site-packages is an antipattern in most containers -- a virtual environment doesn't just isolate your code from other Python code, but from the operating system and other tools (like Poetry) as well. Please see https://github.com/python-poetry/poetry/issues/6397#issuecomment-1236327500 for a pattern you can try in your Dockerfiles, and https://github.com/python-poetry/poetry/issues/6398 for additional context.

**Reply by blnprasad** [reply]
this helps thanks
